### ISSUE TITLE
Deploy application to heal-ctmd.renci.org

### ISSUE THREAD

**Issue by mbwatson** [RENCI/ctmd#131]
So that I don't forget to check back on this, I am noting here that I paused at step 2 (https://github.com/renci/dashboard#install-docker) due to "Network is unreachable" error.

**Comment by mbwatson** [comment]
we don't have heal studies to populate the heal-ctmd dashboard instance with

**Comment by mbwatson** [comment]
@krobasky is going to try this

**Comment by krobasky** [comment]
We're supposed to get the data yesterday.
@dcarsey can you please:
1. Follow-up with Theresa on when the HEAL proposals will be added to redcap and what are the proposal ids
2. create a ticket to track that (if it's not already there), assign to yourself, label with R1b, reference this ticket (e.g., just type 'related to #131' in comment)
3. add comment here with reference to the new ticket
thanks!

**Comment by krobasky** [comment]
@xu-hao I’m deploying the heal-ctmd server and see the following warnings, is this because the server is under-resourced? What are the system requirements for running the ctmd, do you know (maybe this is for @mbwatson )?

> pmd-redis   | 1:M 02 Nov 2019 13:02:57.239 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
> pmd-redis   | 1:M 02 Nov 2019 13:02:57.239 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
>

**Comment by krobasky** [comment]
@mbwatson is this normal?:

```
pmd-frontend-dev | ./src/components/Tables/StudiesTable.js
pmd-frontend-dev |   Line 14:   Expected '===' and instead saw '=='  eqeqeq
pmd-frontend-dev |   Line 101:  No duplicate props allowed           react/jsx-no-duplicate-props
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Tables/SitesTable.js
pmd-frontend-dev |   Line 18:  Expected '===' and instead saw '=='  eqeqeq
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Footer/Footer.js
pmd-frontend-dev |   Line 1:  'useEffect' is defined but never used  no-unused-vars
pmd-frontend-dev |   Line 1:  'useState' is defined but never used   no-unused-vars
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Visualizations/studyEnrollmentGraph.js
pmd-frontend-dev |   Line 263:  Expected '===' and instead saw '=='  eqeqeq
pmd-frontend-dev |   Line 266:  Expected '===' and instead saw '=='  eqeqeq
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Tables/DetailPanels/SiteDetailPanel.js
pmd-frontend-dev |   Line 2:   'makeStyles' is defined but never used                  no-unused-vars
pmd-frontend-dev |   Line 9:   'dateRegPacketSent' is assigned a value but never used  no-unused-vars
pmd-frontend-dev |   Line 9:  'dateSiteActivated' is assigned a value but never used  no-unused-vars
pmd-frontend-dev |
pmd-frontend-dev | ./src/views/Proposals/ByResourcesApproved.js
pmd-frontend-dev |   Line 3:  'Subtitle' is defined but never used  no-unused-vars
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Widgets/DayStats.js
pmd-frontend-dev |   Line 4:  'CardHeader' is defined but never used  no-unused-vars
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Widgets/Counts.js
pmd-frontend-dev |   Line 50:  'theme' is assigned a value but never used  no-unused-vars
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Visualizations/StudyEnrollmentContainer.js
pmd-frontend-dev |   Line 62:  Expected '===' and instead saw '=='  eqeqeq
pmd-frontend-dev |
pmd-frontend-dev | ./src/views/Studies/SitesReport.js
pmd-frontend-dev |   Line 6:   'setFpfv' is assigned a value but never used                    no-unused-vars
pmd-frontend-dev |   Line 7:   'setContractCycleTime' is assigned a value but never used       no-unused-vars
pmd-frontend-dev |   Line 8:   'setIrbCycleTime' is assigned a value but never used            no-unused-vars
pmd-frontend-dev |   Line 9:   'setLpfv' is assigned a value but never used                    no-unused-vars
pmd-frontend-dev |   Line 10:  'setRandomizedPatientRatio' is assigned a value but never used  no-unused-vars
pmd-frontend-dev |   Line 11:  'setExpectedPatientRatio' is assigned a value but never used    no-unused-vars
pmd-frontend-dev |   Line 12:  'setWithdrawnPatientRatio' is assigned a value but never used   no-unused-vars
pmd-frontend-dev |   Line 13:  'setProtocolDeviations' is assigned a value but never used      no-unused-vars
pmd-frontend-dev |   Line 14:  'setQueries' is assigned a value but never used                 no-unused-vars
pmd-frontend-dev |
pmd-frontend-dev | ./src/components/Widgets/ProposalsByTic.js
pmd-frontend-dev |   Line 77:  Array prototype is read only, properties should not be added  no-extend-native
pmd-frontend-dev |
pmd-frontend-dev | ./src/views/Studies/Report.js
pmd-frontend-dev |   Line 78:  Expected '===' and instead saw '=='  eqeqeq
pmd-frontend-dev |
pmd-frontend-dev | ./src/views/Studies/Milestones.js
pmd-frontend-dev |   Line 8:  Array prototype is read only, properties should not be added  no-extend-native
pmd-frontend-dev |
pmd-frontend-dev | Search for the keywords to learn more about each warning.
pmd-frontend-dev | To ignore, add // eslint-disable-next-line to the line before.

```

**Comment by krobasky** [comment]
Instructions complete, but http://heal-ctmd.renci.org isn't working.

**Comment by krobasky** [comment]
Assigned back @mbwatson for feedback, but I'm going to try a few more things and may get it working.

**Comment by krobasky** [comment]
```
added 2024 packages from 916 contributors and audited 39470 packages in 35.563s
found 66 vulnerabilities (63 low, 2 high, 1 critical)
  run `npm audit fix` to fix them, or `npm audit` for details

```
66 vulnerabilities - yikes! should we update npm on ctmd too?

**Comment by krobasky** [comment]
Now I'm out of disk; /var was set-up with only 5GB
I've asked rciadmin for 100GB, hoping that's enough

**Comment by krobasky** [comment]
@mbwatson is there a way for us to update npm in the container you're using so we can fix those 66 vulnerabilities?

**Comment by mbwatson** [comment]
yep. vulnerabilities have been addressed. updates coming in next version.

**Comment by krobasky** [comment]
Matt - what has addressed the vulnerabilities, npm or our repo? Which next version, the npm next version, our repo's next version, or ...?

**Comment by krobasky** [comment]
Deployed! I added '/filter/heal-proposals.csv', and I added 171,186 proposal IDs to test (SPIRRIT and STRESS proposals).

Now we just need to:
- [ ] test that filter is working (@xu-hao, #87)
- [ ] get security audit from Marcin (@dcarsey, #17 )
- [ ] install the data (@dcarsey  #142)

**Comment by mbwatson** [comment]
> Matt - what has addressed the vulnerabilities, npm or our repo? Which next version, the npm next version, our repo's next version, or ...?

@krobasky sorry, i meant that _i_ have addressed the vulnerabilities, and the fix is present in the next dashboard version. i've just deployed it to all three servers.

**Comment by krobasky** [comment]
Confirmed vulnerabilities messages are gone- but I had trouble when I ran as 'su ctmd'; 
I needed to remember to clear the environment so ctmd would pick-up the right version of docker-compose:
`su - ctmd`
(thanks to Hao for the leg-up!)

**Comment by krobasky** [comment]
Checked that data are coming from redcap via logs (thanks @xu-hao ):
`
docker logs pmd-pipeline |&grep "INFO BlockManager"
`
For example:
```
9/11/07 18:09:42 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 7a9a467a2751, 34707, None)
```
A couple of days of log should look something like this:
```
19/11/07 18:09:41 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
19/11/07 18:09:41 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
19/11/07 18:09:42 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
19/11/07 18:09:42 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 7a9a467a2751, 34707, None)
19/11/07 18:09:42 INFO BlockManagerMasterEndpoint: Registering block manager 7a9a467a2751:34707 with 912.3 MB RAM, BlockManagerId(driver, 7a9a467a2751, 34707, None)
19/11/07 18:09:42 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 7a9a467a2751, 34707, None)
19/11/07 18:09:42 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 7a9a467a2751, 34707, None)
19/11/08 00:11:02 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
19/11/08 00:11:02 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
19/11/08 00:11:02 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
19/11/08 00:11:02 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 7a9a467a2751, 34327, None)
19/11/08 00:11:02 INFO BlockManagerMasterEndpoint: Registering block manager 7a9a467a2751:34327 with 912.3 MB RAM, BlockManagerId(driver, 7a9a467a2751, 34327, None)
19/11/08 00:11:02 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 7a9a467a2751, 34327, None)
19/11/08 00:11:02 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 7a9a467a2751, 34327, None)

```

**Comment by krobasky** [comment]
This is blocked on getting advice back from Marcin (#17) so I'm assigning to you, @dcarsey, and you can just assign it back to me after you get info from Marcin.

**Comment by krobasky** [comment]
We got advice from Marcin & opened a couple more tickets that need to be closed before this one can be closed:
In addition to #17, now we also have: #156, #158, #155, #154

**Comment by krobasky** [comment]
I think this is done, assigning to Dawn for internal discussion

**Comment by dcarsey** [comment]
Matt confirms this is done.
